packages/host/app/commands/search-google-images.ts

```diff
@@ -0,0 +1,105 @@
+import { service } from '@ember/service';
+
+import type * as BaseCommandModule from 'https://cardstack.com/base/command';
+
+import HostBaseCommand from '../lib/host-base-command';
+
+import SendRequestViaProxyCommand from './send-request-via-proxy';
+
+import type CommandService from '../services/command-service';
+
+export default class SearchGoogleImagesCommand extends HostBaseCommand<
+  typeof BaseCommandModule.SearchGoogleImagesInput,
+  typeof BaseCommandModule.SearchGoogleImagesResult
+> {
+  @service declare private commandService: CommandService;
+
+  static actionVerb = 'Search Google Images';
+  description = 'Search for images on Google using the Custom Search API';
+
+  async getInputType() {
+    let commandModule = await this.loadCommandModule();
+    const { SearchGoogleImagesInput } = commandModule;
+    return SearchGoogleImagesInput;
+  }
+
+  protected async run(
+    input: BaseCommandModule.SearchGoogleImagesInput,
+  ): Promise<BaseCommandModule.SearchGoogleImagesResult> {
+    const commandModule = await this.loadCommandModule();
+    const { SearchGoogleImagesResult } = commandModule;
+
+    try {
+      const maxResults = input.maxResults || 10;
+      const startIndex = input.startIndex || 1;
+      const searchQuery = encodeURIComponent(input.query);
+
+      // Build the Google Custom Search API URL
+      // The API key and search engine ID will be handled by the proxy endpoint
+      const searchUrl = `https://www.googleapis.com/customsearch/v1?q=${searchQuery}&searchType=image&num=${Math.min(maxResults, 10)}&start=${startIndex}&cx=064501294a5c9430a`;
+
+      const sendRequestViaProxyCommand = new SendRequestViaProxyCommand(
+        this.commandService.commandContext,
```

Comment:
```suggestion
        this.commandContext,
```

And remove the `@service commandService` line. This means that this command doesn't have to extend from host command